modules/@shopify/checkout-sheet-kit/src/index.ts

```diff
@@ -78,63 +81,125 @@ class ShopifyCheckoutSheet implements ShopifyCheckoutSheetKit {
     event: CheckoutEvent,
     callback: CheckoutEventCallback,
   ): EmitterSubscription | undefined {
-    if (event === 'pixel' && typeof callback === 'function') {
-      const eventHandler = callback as (data?: PixelEvent) => void;
-
-      /**
-       * Event data can be sent back as either a parsed Pixel Event object or a JSON string.
-       */
-      const cb = (eventData: string | PixelEvent) => {
-        try {
-          if (typeof eventData === 'string') {
-            try {
-              const parsed = JSON.parse(eventData);
-
-              if (
-                parsed.hasOwnProperty('customData') &&
-                typeof parsed.customData === 'string'
-              ) {
-                try {
-                  parsed.customData = JSON.parse(parsed.customData);
-                } catch {}
-              }
-              eventHandler(parsed as PixelEvent);
-            } catch (error) {
-              const parseError = new WebPixelsParseError(
-                'Failed to parse Web Pixel event data: Invalid JSON',
-                {
-                  cause: 'Invalid JSON',
-                },
-              );
-              // eslint-disable-next-line no-console
-              console.error(parseError);
-            }
-          } else if (eventData && typeof eventData === 'object') {
-            eventHandler(eventData);
-          }
-        } catch (error) {
-          const parseError = new WebPixelsParseError(
-            'Failed to parse Web Pixel event data',
-            {
-              cause: 'Unknown',
-            },
-          );
-          // eslint-disable-next-line no-console
-          console.error(parseError);
-        }
-      };
-
-      // Web Pixel event specific handler
-      return ShopifyCheckoutSheet.eventEmitter.addListener(event, cb);
+    let eventCallback: CheckoutEventCallback = callback;
+
+    switch (event) {
+      case 'pixel':
+        eventCallback = this.handleWebPixelEvent(
+          callback,
+        ) as PixelEventCallback;
+        break;
+      case 'completed':
+        eventCallback = this.handleCompletedEvent(
+          callback,
+        ) as CheckoutCompletedEventCallback;
+        break;
+      default:
+        eventCallback = callback;
     }
 
     // Default handler for all non-pixel events
-    return ShopifyCheckoutSheet.eventEmitter.addListener(event, callback);
+    return ShopifyCheckoutSheet.eventEmitter.addListener(event, eventCallback);
   }
 
   public removeEventListeners(event: CheckoutEvent) {
     ShopifyCheckoutSheet.eventEmitter.removeAllListeners(event);
   }
+
+  // ---
+
+  private handleWebPixelEvent(
+    callback: CheckoutEventCallback,
```

Comment:
Why is the type `CheckoutEventCallback` when we always cast it to `PixelEventCallback` in the first line?